### DISCUSSION TITLE
"monaco is not defined" after upgrading monaco-editor from 0.20.0 to 0.31.1

### DISCUSSION THREAD

**Discussion by chengtie** [eclipse-theia/theia#10639] [General]
I have been using "monaco-languageclient": "^0.13.0", "monaco-editor": "^0.20.0" and "react-monaco-editor": "^0.35.0" in a project. I used `import { MonacoToProtocolConverter, ProtocolToMonacoConverter } from 'monaco-languageclient/lib/monaco-converter'` in my code, the two functions worked great.

I just upgraded Monaco Editor to "^0.31.1", then the project raised an error:

![Screenshot 2022-01-16 at 11 30 58](https://user-images.githubusercontent.com/774409/149656408-3ae04687-c9b5-4969-ab6f-f81da6b14f90.png)

I tried also other versions of Monaco Editor (by eg, using `npm install monaco-editor@0.25.0`), the above error started from Monaco Editor 0.23.0. Here is a part of `yarn.lock`:

```
monaco-editor@*, monaco-editor@^0.23.0:
  version "0.23.0"
  resolved "https://registry.yarnpkg.com/monaco-editor/-/monaco-editor-0.23.0.tgz#24844ba5640c7adb3a2a3ff3b520cf2d7170a6f0"
  integrity sha512-q+CP5zMR/aFiMTE9QlIavGyGicKnG2v/H8qVvybLzeFsARM8f6G9fL0sMST2tyVYCwDKkGamZUI6647A0jR/Lg==

monaco-languageclient@^0.13.0:
  version "0.13.0"
  resolved "https://registry.yarnpkg.com/monaco-languageclient/-/monaco-languageclient-0.13.0.tgz#59b68b42fb7633171502d6557f597c2752f6c266"
  integrity sha512-aCwd33dTitwV5QwY56rpYHwzEGXei8TZ+yvZcvP3gEMd6Mizr8m3pOuoknDi2SUfLuNAHS6+ulvLgZlNQB5awg==
  dependencies:
    glob-to-regexp "^0.3.0"
    vscode-jsonrpc "^5.0.0"
    vscode-languageclient "^6.0.0"
    vscode-uri "^2.1.1"

react-monaco-editor@^0.35.0:
  version "0.35.0"
  resolved "https://registry.yarnpkg.com/react-monaco-editor/-/react-monaco-editor-0.35.0.tgz#ceaaca4f292571e95957871145bd7c5f76a4fd33"
  integrity sha512-iQbJykYPgKko0yqphTVWi1vHFokFShv5qpWnFV2TvtliKG45x2LVF4/HC06MDVK4fBfp2lCrun+NUcEYMzAnNQ==
  dependencies:
    "@types/react" "^15.x || ^16.x"
    monaco-editor "*"
    prop-types "^15.7.2"
```

Does anyone know what may be the cause?

**Comment by vince-fugnitto** [comment]
@chengtie thank you for the discussion, are you trying to use `monaco-editor` in the `theia` framework?

The framework does not rely on `monaco-editor` directly but through a custom build of vscode which we generate and publish `@theia/monaco-editor-core`, and we currently use `0.23.x`.
